src/utilities/transition/BaseTransition.js

```diff
@@ -0,0 +1,250 @@
+'use strict';
+
+// Required modules.
+var EventObserver = require( '../../mixins/Events.js' );
+var fnBind = require( '../function-bind' ).bind;
+
+/**
+ * BaseTransition
+ * @class
+ *
+ * @classdesc Initializes new BaseTransition behavior.
+ *   This shouldn't be used directly, but instead should be
+ *   the base class used through composition by a specific transition.
+ *
+ * @param {HTMLNode} element
+ *   DOM element to apply transition to.
+ * @param {Object} classes
+ *   The classes to apply to this transition.
+ * @returns {BaseTransition} An instance.
+ */
+function BaseTransition( element, classes ) { // eslint-disable-line max-statements, no-inline-comments, max-len
+  var _classes = classes;
+  var _dom;
+
+  var _lastClass;
+  var _transitionEndEvent;
+  var _transitionCompleteBinded;
+  var _addEventListenerBinded;
+  var _isAnimating = false;
+  var _isFlushed = false;
+
+  /**
+   * @returns {BaseTransition} An instance.
+   */
+  function init() {
+    _transitionCompleteBinded = fnBind( _transitionComplete, this );
+    _addEventListenerBinded = fnBind( _addEventListener, this );
+    setElement( element );
+
+    return this;
+  }
+
+  /**
+   * Set the HTML element target of this transition.
+   * @param {HTMLNode} elem - The target of the transition.
+   */
+  function setElement( elem ) {
+    // If the element has already been set,
+    // clear the transition classes from the old element.
+    if ( _dom ) {
+      remove();
+      animateOn();
+    }
+    _dom = elem;
+    _dom.classList.add( _classes.BASE_CLASS );
+    _transitionEndEvent = _getTransitionEndEvent( _dom );
+  }
+
+  /**
+   * Add a "transition-duration: 0s" utility CSS class.
+   * @returns {BaseTransition} An instance.
+   */
+  function animateOn() {
+    if ( !_dom ) { return this; }
+    _dom.classList.remove( BaseTransition.NO_ANIMATION_CLASS );
+
+    return this;
+  }
+
+  /**
+   * Remove a "transition-duration: 0s" utility CSS class.
+   * @returns {BaseTransition} An instance.
+   */
+  function animateOff() {
+    if ( !_dom ) { return this; }
+    _dom.classList.add( BaseTransition.NO_ANIMATION_CLASS );
+
+    return this;
+  }
+
+  /**
+   * @returns {boolean} Whether the transition has a duration or not.
+   *   Returns false if this transition has not been initialized.
+   */
+  function isAnimated() {
+    if ( !_dom ) { return false; }
+    return !_dom.classList.contains( BaseTransition.NO_ANIMATION_CLASS );
+  }
+
+  /**
+   * Halt an in-progress animation and call the complete event immediately.
+   */
+  function halt() {
+    if ( !_isAnimating ) { return; }
+    _dom.style.webkitTransitionDuration = '0';
```

Comment:
I'm leaving this as-is because I don't know the best way to accomplish what @sebworks suggests, and I don't see it as a major issue (and I'm in a hurry 😄 ).